src/builds/builds.controller.ts

```diff
@@ -75,7 +75,7 @@ export class BuildsController {
     await this.buildsService.deleteOldBuilds(project.id, project.maxBuildAllowed);
     const build = await this.buildsService.findOrCreate({
       projectId: project.id,
-      branchName: createBuildDto.branchName,
+      branchName: (createBuildDto.branchName && createBuildDto.branchName.trim().length > 0) ? createBuildDto.branchName : project.mainBranchName,
```

Comment:
empty branch name should be validated in another place if needed
```suggestion
      branchName: createBuildDto.branchName ?? project.mainBranchName,
```

Reply:
please, cover this with e2e test https://github.com/Visual-Regression-Tracker/backend/blob/master/test/builds.e2e-spec.ts

Reply:
Thanks. Added. Do you think we should make branchName optional in the DTO just like ciBuildId ?